### PR TITLE
test(tree2): More representative perf tests

### PR DESCRIPTION
**Pull request by yann-achard-MS**
Makes the `EditManager - Bench` test suite to run with the `ChangeFamily` that is used in production (in addition to the no-op rebaser).

### CODE REVIEW

File: experimental/dds/tree2/src/test/editMinter.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export type Editor = (builder: DefaultEditBuilder) => void;
+
+let builtChangeset: DefaultChangeset | undefined;
+
+export function makeEditMinter(
+	family: DefaultChangeFamily,
+	editor: Editor,
+): () => DefaultChangeset {
+	const innerEditor = family.buildEditor((change) => {
+		assert(builtChangeset === undefined);
```
Comment (Abe27342) [inline experimental/dds/tree2/src/test/editMinter.ts:18]: is it intentional that `builtChangeset` is scoped to the module and not inside `makeEditMinter`? Seems weird to me that creating multiple edit minters and interleaving their calls shares this state.

File: experimental/dds/tree2/src/test/shared-tree-core/editManagerTestUtils.ts [hunk 2]
```diff
@@ -43,34 +53,36 @@ export function editManagerFactory(options: {
 			revertible.discard();
 		});
 	}
-
-	return { manager, family };
+	return manager;
 }
 
-export function rebaseLocalEditsOverTrunkEdits(
+export function rebaseLocalEditsOverTrunkEdits<TChange>(
```
Comment (Abe27342) [inline experimental/dds/tree2/src/test/shared-tree-core/editManagerTestUtils.ts:59]: preexisting, but do we even get anything from overloading here and below? We're not documenting the different overloads with different doc comments, and just using an optional param with default `false` seems about as clear.

**Review by Abe27342 — APPROVED** [review]
looks good minus the two nits

File: experimental/dds/tree2/src/test/shared-tree-core/editManagerTestUtils.ts (hunk 2 above)
Comment (yann-achard-MS) [reply experimental/dds/tree2/src/test/shared-tree-core/editManagerTestUtils.ts:59]: We do get something out of it: if you pass the `defer` argument then you're guaranteed to get the `() => void` back, and if you don't pass it, then you're guaranteed not to. Adding doc comments to help.

File: experimental/dds/tree2/src/test/editMinter.ts (hunk 1 above)
Comment (yann-achard-MS) [reply experimental/dds/tree2/src/test/editMinter.ts:18]: It didn't occur to me to scope it in but that's a good idea. :)

**Pull request merged**